lib/msf/core/post/file.rb

```diff
@@ -704,6 +704,61 @@ def _unix_max_line_length
     line_max
   end
 
+  def search( root=nil, glob='*.*', recurse=true, timeout=-1 )
+    matches = []
+
+    if session.type == 'meterpreter'
+      return session.fs.file.search(root, glob, recurse, timeout)
+
+    elsif session.type == 'powershell'
+      list = cmd_exec("Get-ChildItem #{recurse ? '-Recurse': ''} -Path #{root}| where {! $_.PSIsContainer} | Format-Table Name, Length, Directory").split("\n")
+      list.each do |file_info|
+        file_info = file_info.split
+        attrib = {}
+        attrib['name'] = file_info[0]
+        next unless attrib['name'].match? (glob)
+        attrib['size'] = file_info[1]
+        attrib['path'] = file_info[2]
+        matches << attrib
+      end
+
+    elsif session.platform == 'windows'
+      #cmdcode  
+    else
+      if recurse
+        list = cmd_exec("ls -la1Rp #{root} | grep -v /$").split("\n\n")
+        list.each do |each_dir|
+          path, files = each_dir.split(":\n")
+          files = files.split("\n")
+          files.delete_at(0)
+          files.each do |file_info|
+            file_info = file_info.split
+            attrib = {}
+            attrib['name'] = file_info[8..-1].join('')
+            next unless attrib['name'].match? (glob)
+            attrib['size'] = file_info[4]
+            attrib['path'] = path
+            matches << attrib
+          end
+        end
+      else
+        list = cmd_exec('ls -la1p  | grep -v /$').split("\n")
+        list.delete_at(0)
+        list.each do |file_info|
+          file_info = file_info.split
+          attrib = {}
+          attrib['name'] = file_info[8..-1].join('')
+          next unless attrib['name'].match? (glob)
+          attrib['size'] = file_info[4]
+          attrib['path'] = root
+          matches << attrib
+        end
+      end
+    end
```

Comment:
These blobs are understandably larger than the other because you're having to handle it all yourself for shell sessions. Would you be mind breaking it out into a sub function so it's not nested quite so deep? Something like `shell_search` like the other mixins use would be nice.

Reply:
Now it's little smaller because I've replaced `ls -l1pa` with `find` command. But `find` command might not be present in every system so should I create a new method which uses `ls -l1pa`  to which the method should fall to in case `find` is not present?